File: Games_Factory_CRUD_Kotlin/app/src/main/java/edu/raultirado/games_factory_crud_kotlin/data/model/Usuario.kt
package edu.raultirado.games_factory_crud_kotlin.data.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "Usuario")
data class Usuario(
    @PrimaryKey
    @ColumnInfo(name = "ID_DNI")
    val idDni: String = "",

    @ColumnInfo(name = "nombre_usu")
    val nombreUsu: String = "",

    @ColumnInfo(name = "apellidos_usu")
    val apellidosUsu: String = "",

    val direccion: String = "",

    // CAMBIO: De Date a String
    @ColumnInfo(name = "fecha_naci")
    val fechaNaci: String = "",

    val telefono: String = "",

    @ColumnInfo(name = "codigo_postal")
    val codigoPostal: String = "",

    @ColumnInfo(name = "correo_usu")
    val correoUsu: String = "",

    @ColumnInfo(name = "contrasena_usu")
    val contrasenaUsu: String = "",
)

File: Games_Factory_WPF/Models/Usuario.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Games_Factory.Models
{
    public class Usuario
    {
        [Key]
        [Column("ID_DNI")]
        [StringLength(11)]
        public string IdDni { get; set; } = null!;

        [Column("nombre_usu")]
        [StringLength(30)]
        public string Nombre { get; set; } = null!;

        [Column("apellidos_usu")]
        [StringLength(50)]
        public string Apellidos { get; set; } = null!;

        [Column("direccion")]
        [StringLength(55)]
        public string Direccion { get; set; } = null!;

        [Column("fecha_naci")]
        public DateTime FechaNacimiento { get; set; }

        [Column("telefono")]
        [StringLength(9)]
        public string Telefono { get; set; } = null!;

        [Column("codigo_postal")]
        [StringLength(5)]
        public string CodigoPostal { get; set; } = null!;

        [Column("correo_usu")]
        [StringLength(55)]
        public string Correo { get; set; } = null!;

        [Column("contrasena_usu")]
        [StringLength(150)]
        public string Contrasena { get; set; } = null!;

        // Navegación
        public virtual ICollection<UsuarioProducto> Compras { get; set; } = new List<UsuarioProducto>();
    }
}
